sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -125,10 +125,18 @@ def _build_create_table_exp(
             # is applied to a ctas statement, VARCHAR types default to 1 in some instances.
             # this checks the explain plain from redshift and tries to detect when these optimizer
             # bugs occur and force a cast
-            sql = statement.sql(
+            explain_statement = statement.copy()
+            for select in explain_statement.find_all(exp.Select):
+                if select.args.get("from"):
+                    select.args["limit"] = None
+                    select.args["where"] = None
```

Comment:
Nit(style): we usually do this for popping off args that may not exist in sqlglot:
```suggestion
                    select.set("limit", None)
                    select.set("where", None)
```